### ISSUE TITLE
User - COVID-19 Status - Agree to Share Locations to Server

### ISSUE THREAD

**Issue by iamjoshfrank** [revelry-foundation/nightingale#46]
_Sibling of #41 and #42

## Background

_**Share my location**_ isn't needed until a user indicates they have tested positive. We currently have no use / no desire to gather the location data of negative/unknown-status users.

Flow should look something like this for sharing location history to the server:

1. `Change status to positive`
2. `Supply additional information (dates for test / test result / start of symptoms)`
3. `Request the user agree to share locations for the relevant time period they were likely contagious`

**Comment by iamjoshfrank** [comment]
"Relevant time period" should equal the contact elicitation period, eg the period of time the positive-status user was likely infectious. See my [comment on #44](https://github.com/revelry-foundation/nightingale/issues/44#issuecomment-635673271), I'm not sure where this should land. Probably here, since we don't want to upload a bunch of irrelevant location history - only the list of locations that fall within the infectious period.

**Comment by CurtisVonRubenhoff** [comment]
I think this ticket falls under the purview of what I'm working on (API endpoint to make this possible). I self-assigned this while I'm working, but it won't be done until it's implemented in the mobile app (maybe i can shift my focus to that unless someone currently on the mobile side is itching to grab this). 

I should have a pr for the API endpoint within the hour.

**Comment by iamjoshfrank** [comment]
Awesome, thanks @CurtisVonRubenhoff !

I'll be back this way in ~20 minutes to finish off making the Scenario and Acceptance Criteria crystal clear re: the date range for which the app should share locations based on test and/or symptom dates.

I'm assuming that work (that logic?) would happen on the mobile end; and that the API only serves as a landing spot for the selected location data. Yeah?

**Comment by CurtisVonRubenhoff** [comment]
That's accurate. This endpoint is only ingesting locations and when you were there

**Comment by Mcscott142** [comment]
Wires: 

**Onboarding:** 
<img width="432" alt="Screen Shot 2020-05-28 at 5 56 33 PM" src="https://user-images.githubusercontent.com/49654454/83202140-93d34f00-a10c-11ea-8797-4e90e1d3ca23.png">

<img width="555" alt="Screen Shot 2020-05-29 at 9 53 44 AM" src="https://user-images.githubusercontent.com/49654454/83275525-dfc9d680-a194-11ea-9af8-c5209324ee42.png">

<img width="176" alt="Screen Shot 2020-05-29 at 10 08 44 AM" src="https://user-images.githubusercontent.com/49654454/83275537-e22c3080-a194-11ea-9469-72bcc39d3f6d.png">

**Comment by iamjoshfrank** [comment]
Update! Based on conversations in Slack spurred by @grossvogel [here](https://revelry-public.slack.com/archives/C014AR56T1P/p1590764962152300?thread_ts=1590764499.147300&cid=C014AR56T1P) and [here](https://revelry-public.slack.com/archives/C014AR56T1P/p1590765480157000) re: **trade-offs** and **decisions* when we're trying to ship a Proof of Concept in under a week:

1. We're going to postpone the more complicated (but much more accurate) UI of entering dates for test results and onset of symptoms in order to filter down location dates for submission.
2. For now, we'll upload the prior 14 days of locations from the date the user changes their status from "negative" to "positive" (still privately, anonymously, and only with their discrete consent).

Spinning up a clean Issue for ^^ and I'm going to move this Issue along with #42 into a future Milestone.

**Comment by CurtisVonRubenhoff** [comment]
Nope. I think I'm still good. I'll move my updates to #50 in that case though
